build.gradle

```diff
@@ -170,6 +170,17 @@ if (bwc_tests_enabled == false) {
   println "See ${bwc_tests_disabled_issue}"
   println "==========================================================="
 }
+if (project.gradle.startParameter.taskNames.find { it.startsWith("checkPart") } != null) {
```

Comment:
This kind of thing tends to be quite brittle. One major issues is that "tasks that will execute" != "tasks you asked to execute". For example, if I introduce a task `checkAllParts` which `dependsOn checkPart1, checkPart1` then this breaks down.

If we don't want tasks to run as part of a build we should define new tasks with the correct dependencies. Introducing logic like this into the build makes it very difficult to reason about why a given task was executed (or not) during a particular build invocation. In the end we should strive for more specific lifecycle tasks, not more dynamic logic. If I don't want task `foo` to run when I run `checkPart`, then `checkPart` shouldn't depend on `foo`. If making this so means adding more tasks and tweaking the existing dependency graph then we should do that instead.

Reply:
Splitting the build up is not (yet) something we want to do, so we wouldn't want to embed this into every project. In the end, setting the bwc tests to false does prevent the dependencies to be added to check, 
I'm not overly concerned about tasks that would depend on these partial checks as these are done specifically for PR checks in CI, maybe we should even reflect that in the name of the task to make it clear. I would be much more concerned if it would be possible to accidentally run fewer tests, running more than one intended isn't as bad.